packages/integration_test/lib/src/reporter.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:test_core/src/runner/engine.dart';
+import 'package:test_core/src/runner/reporter.dart';
+
+import '../common.dart';
+import 'constants.dart';
+
+/// A reporter that plugs into [directRunTests] from `package:test_core`.
+class ResultReporter implements Reporter {
+  final Engine _engine;
+  Completer<Map<String, Object>> _resultsCompleter;
+
+  final _subscriptions = <StreamSubscription<Object>>{};
+
+  /// When the [_engine] has completed execution of tests, [_resultsCompleter]
+  /// will be completed with the test results.
+  ResultReporter(this._engine, this._resultsCompleter) {
+    _subscriptions.add(_engine.success.asStream().listen(_onDone));
+  }
+
+  void _onDone(bool _) async {
+    _cancel();
+    final Map<String, Object> results = {
+      for (final liveTest in _engine.liveTests)
+        liveTest.test.name: liveTest.state.result.name == success
+            ? success
+            : Failure(
+                liveTest.test.name, liveTest.errors.first.stackTrace.toString(),
```

Comment:
Is the first stack trace the right one? 

Is the first error the right one?

Is there some way we could add _all_ stack traces and errors to the reporting?

Reply:
This was taken from the way other Dart tests are integrated with the Google3 infrastructure. I'm not sure if we should do anything else, @natebosch would you know more about this?